query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLInstantQuery.java

```diff
@@ -0,0 +1,16 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.time.Instant;
+import java.util.List;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+
+@Getter
+@Builder
+class PromQLInstantQuery {
+  @NonNull @Singular private List<String> queries;
+
+  @NonNull private Instant evalTime;
```

Comment:
I think I'm confusing PRs at this point, wasn't this part of the other one? Anyway:
- what is eval time?
- Why does query contain queries. Probably need some naming changes, I get that one QS query may map to multiple promql queries, but the class itself is called a promql query.
- An instant, as explained to me elsewhere, is an aggregation - an aggregation still should require a time range.
- Please use `@Value` everywhere.

Reply:
Yes, both the PRs (the other one - https://github.com/hypertrace/query-service/pull/110) need the same object to proceed. We could have merged these Query-related object classes first. So, now whichever is get merged, others have to use the same one.

> what is eval time?
Eval Time is the time of instant when the metrics queries are evaluated.
It is one of the query param arguments of PromQL Instant query API : https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries

In the Query Service context, for a given time range filter, it is a higher range of the time filter.
 e.g start_time_millis >= lower AND start_time_millis < higher

The time range will be translated to `[]` promQL expression.

```
sum by (backend_id) 
(sum_over_time(num_calls{tenant_id="__default"}[3600000ms])) 
@ 1631181508474
```
As in the above example, see [3600000ms] that's your time filter of QS's aggregation query (difference of higher - lower value of time range filter). And @ 1631181508474 is a higher range of time filter is evalTime for promQL query.

```
filter {
  childFilter {
    childFilter {
      lhs {
        columnIdentifier {
          columnName: "BACKEND.startTime"
        }
      }
      operator: GE
      rhs {
        literal {
          value {
            valueType: LONG
            long: 1631177908474
          }
        }
      }
    }
    childFilter {
      lhs {
        columnIdentifier {
          columnName: "BACKEND.startTime"
        }
      }
      operator: LT
      rhs {
        literal {
          value {
            valueType: LONG
            long: 1631181508474
          }
        }
      }
    }
  }
}

```

> Why does query contain queries. Probably need some naming changes, I get that one QS query may map to multiple promql queries, but the class itself is called a promql query.

Yes, it is because a single QS query can map to multiple promQL query. Should we rename that as `PromQLInstantQueries`

Reply:
Thanks for the second explanation, sorry to make you repeat yourself :)

> Yes, it is because a single QS query can map to multiple promQL queries. Should we rename that as PromQLInstantQueries?

Yeah that makes sense. It feels a bit funky, but I don't have a better recommendation.